File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        int maxHeight = 0;
        int maxWidth = 0;
        // 가로는 size[0] 세로는 size[1]
        for(int[] size: sizes) {
             // 각 명함을 회전해서 큰 값은 가로, 작은 값은 세로
            int width = Math.max(size[0], size[1]);
            int height = Math.min(size[0], size[1]);
            // System.out.println(width); // 6,7,6,8
            // System.out.println(height); // 5,3,3,4
            
            maxWidth = Math.max(maxWidth, width);
            maxHeight = Math.max(maxHeight, height);
        }
        
        return maxHeight * maxWidth;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.kt
class Solution {
    fun solution(sizes: Array<IntArray>): Int {
        var maxWidth = 0
        var maxHeight = 0
        var answer = 0
        
         // 각 명함을 순회하면서 가로와 세로를 정렬
        for(size in sizes) {
            // 가로와 세로 중 더 큰 값을 가로로 설정
            val width = maxOf(size[0], size[1])
            // 가로와 세로 중 더 작은 값을 세로로 설정
            val height = minOf(size[0], size[1])
            // println(width)
            // println(height)
            maxWidth = maxOf(maxWidth, width)
            // println(maxWidth)
            maxHeight = maxOf(maxHeight, height)
            // println(answer)
        }
        answer = maxHeight * maxWidth
        
        return answer
    }
}

//